src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/external/PoPUserApiService.kt

```diff
@@ -44,4 +45,14 @@ class PoPUserApiService(
       throw ResourceNotFoundException("Person On Probation User verification already done.")
     }
   }
+
+  fun getVerifiedPopUsers(): List<uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.data.popuserapi.PopUserData> {
+    return popUserWebClientCredentials
+      .get()
+      .uri("/person-on-probation-user/users/all")
+      .retrieve()
+      .bodyToFlux<uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.data.popuserapi.PopUserData>()
+      .collectList()
+      .block() ?: throw RuntimeException("Unexpected null returned from request.")
```

Comment:
It would be better to throw a custom type of exception, I'll check the code to see if there is one more suitable

Reply:
perhaps ResourceNotFoundException or NoDataWithCodeFoundException

Reply:
oh nevermind actually I see this is a pattern everywhere in the code, .block() ?: throw Runtime, best to leave it as is then for consistency

Reply:
ok - no change